tests/lib/linter/file-report.js

```diff
@@ -0,0 +1,2004 @@
+				fileReport.messages[0].suggestions[0].fix,
+				additionalFix,
+			);
+			assert.notStrictEqual(
+				fileReport.messages[0].suggestions[0].fix.range,
+				additionalFix.range,
+			);
+		}
+
+		it("should deep clone returned fix object", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				fix: () => fix,
+			});
+
+			assertFixMatches(fileReport);
+		});
+
+		it("should deep clone returned fix object", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				fix: () => fix,
+			});
+
+			assertFixMatches(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when `fix()` returns an array with a single item", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				fix: () => [fix],
+			});
+
+			assertFixMatches(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when `fix()` returns an array with multiple items", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				fix: () => [fix, additionalFix],
+			});
+
+			assertAdditionalFixNoMatch(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when `fix()` generator yields a single item", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				*fix() {
+					yield fix;
+				},
+			});
+
+			assertFixMatches(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when `fix()` generator yields multiple items", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				*fix() {
+					yield fix;
+					yield additionalFix;
+				},
+			});
+
+			assertAdditionalFixNoMatch(fileReport);
+			assertAdditionalFixNoMatch(fileReport);
+		});
+
+		it("should deep clone returned suggestion fix object", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				suggest: [
+					{
+						messageId: "suggestion1",
+						fix: () => fix,
+					},
+				],
+			});
+
+			assertSuggestionFixMatches(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when suggestion `fix()` returns an array with a single item", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				suggest: [
+					{
+						messageId: "suggestion1",
+						fix: () => [fix],
+					},
+				],
+			});
+
+			assertSuggestionFixMatches(fileReport);
+		});
+
+		it("should create a new fix object with a new range array when suggestion `fix()` returns an array with multiple items", () => {
+			fileReport.addRuleMessage("foo-rule", 2, {
+				node,
+				messageId: "testMessage",
+				suggest: [
+					{
+						messageId: "suggestion1",
+						fix: () => [fix, additionalFix],
+					},
+				],
+			});
+
+			assertSuggestionFixNoMatch(fileReport);
```

Comment:
```suggestion
			assertSuggestionFixNoMatch();
```

Also missing check for `additionalFix`.